packages/fast-layouts-react/src/grid/grid.tsx

```diff
@@ -0,0 +1,160 @@
+        tag: GridTag.div,
+        gridColumn: 2,
+        gutter: 8,
+        verticalAlign: GridAlignment.stretch,
+        horizontalAlign: GridAlignment.stretch
+    };
+
+    protected handledProps: GridProps = {
+        columnCount: void 0,
+        gridColumn: void 0,
+        gutter: void 0,
+        horizontalAlign: void 0,
+        managedClasses: void 0,
+        tag: void 0,
+        verticalAlign: void 0
+    };
+
+    /**
+     * Renders the Grid markup
+     */
+    public render(): JSX.Element {
+        return (
+            <this.tag
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                style={this.generateStyleAttributes()}
+            >
+                {this.renderChildren()}
+            </this.tag>
+        );
+    }
+
+    /**
+     * Component has mounted
+     */
+    public componentDidMount(): void {
+        if (this.shouldTrackBreakpoints(this.props)) {
+            BreakpointTracker.subscribe(this.update);
+        }
+    }
+
+    /**
+     * Component will be unmounted
+     */
+    public componentWillUnmount(): void {
+        BreakpointTracker.unsubscribe(this.update);
+    }
+
+    /**
+     * Component has updated
+     */
+    public componentDidUpdate(previousProps: GridProps): void {
+        if (this.shouldTrackBreakpoints(this.props) && !this.shouldTrackBreakpoints(previousProps)) {
+            // If we should be tracking breakpoints but previously weren't, then subscribe to changes
+            BreakpointTracker.subscribe(this.update);
+        } else if (!this.shouldTrackBreakpoints(this.props) && this.shouldTrackBreakpoints(previousProps)) {
+            // If we were tracking breakpoints but we shouldn't be now, then unsubscribe from chnages
+            BreakpointTracker.unsubscribe(this.update);
+        }
+    }
+
+    protected generateClassNames(): string {
+        const className: string = this.props.managedClasses.grid;
+
+        return super.generateClassNames(className);
+    }
+
+    /**
+     * Determines if we should be tracking breakpoints based on a set of props
+     */
+    private shouldTrackBreakpoints(props: GridProps): boolean {
+        return Array.isArray(props.gutter);
+    }
+
+    /**
+     * Generates the column-span value
+     */
+    private generateGutter(): number {
+        if (typeof this.props.gutter === "number") {
+            return this.props.gutter;
+        } else if (Array.isArray(this.props.gutter)) {
+            return getValueByBreakpoint(this.props.gutter);
+        } else {
+            return null;
+        }
+    }
+
+    private generateAlignment(alignment: GridAlignment): string {
+        return GridAlignment[alignment];
+    }
+
+    /**
+     * Force the component to update
+     */
+    private update = (): void => {
+        this.forceUpdate();
+    }
+
+    private generateStyleAttributes(): React.HTMLAttributes<HTMLDivElement> {
```

Comment:
similar to column can we combine any given unhandled props into the returned style?

Reply:
yep - good idea.